src/cloudant/replicator.py

```diff
@@ -60,27 +60,46 @@ def create_replication(self, source_db=None, target_db=None,
 
         :returns: Replication document as a Document instance
         """
+        if source_db is None:
+            raise CloudantReplicatorException(101)
+
+        if target_db is None:
+            raise CloudantReplicatorException(102)
 
         data = dict(
             _id=repl_id if repl_id else str(uuid.uuid4()),
             **kwargs
         )
 
-        if source_db is None:
-            raise CloudantReplicatorException(101)
+        # replication source
+
         data['source'] = {'url': source_db.database_url}
-        if not source_db.admin_party:
-            data['source'].update(
-                {'headers': {'Authorization': source_db.creds['basic_auth']}}
-            )
+        if source_db.admin_party:
+            pass  # no credentials required
+        elif source_db.client.is_iam_authenticated:
```

Comment:
The auth used in a replicator document should be independent of the method with which the client is authenticated. For example there is no reason why I can't login python-cloudant with my account credentials and create a replicator document using any other credentials including IAM API keys.

This code should be a fallback only if the `source_user` is empty and target_user are empty. I think however, that a new version of the `create_replication` should require some credential information and the fallback should only be used in the backwards compatibility case.

Reply:
`source_db` and `target_db` are supplied to the `create_replication` method. They are type database objects and therefore don't share credentials used to POST the replication doc.

Reply:
Yes, I see what you mean, everything is an object, so the specific credentials will have been supplied up front in construction of the necessary source or target database objects (or more specifically the client instances needed for those if they are distinct from the replicator.

I think it would be nice if a replication could be created with a source & target URLs and the associated API keys i.e. 4 strings instead of needing to potentially instantiate multiple clients and DBs, but that is out of scope here.

Reply:
Yep, that's why I think it is out of scope to try and re-introduce that in a working fashion here.